src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -92,44 +95,63 @@ const UsersFragment = ({
     if (from === FROM_FOLLOWED_USERS_INITIAL || from === FROM_UNFOLLOWED_USERS_INITIAL) {
       const newFollowedUsers = [...users];
       exhangeFollower(newFollowedUsers, willFollow, index);
-
       DiscoveryAction.setDiscoveryInitialUsers(newFollowedUsers, discoveryDispatch);
+      if (!netInfo.isConnected)
+        setTimeout(() => {
+          exhangeFollower(newFollowedUsers, !willFollow, index);
+          DiscoveryAction.setDiscoveryInitialUsers(newFollowedUsers, discoveryDispatch);
+        }, 2000);
     }
 
     if (from === FROM_FOLLOWED_USERS) {
       const newFollowedUsers = [...followedUsers];
       exhangeFollower(newFollowedUsers, willFollow, index);
-
       setFollowedUsers(newFollowedUsers);
+      if (!netInfo.isConnected)
+        setTimeout(() => {
+          exhangeFollower(newFollowedUsers, !willFollow, index);
+          setFollowedUsers(newFollowedUsers);
+        }, 2000);
     }
 
     if (from === FROM_UNFOLLOWED_USERS) {
       const newUnfollowedUsers = [...unfollowedUsers];
       exhangeFollower(newUnfollowedUsers, willFollow, index);
       setUnfollowedUsers(newUnfollowedUsers);
+      if (!netInfo.isConnected)
+        setTimeout(() => {
+          exhangeFollower(newUnfollowedUsers, !willFollow, index);
+          setUnfollowedUsers(newUnfollowedUsers);
+        }, 2000);
     }
 
     if (from === FROM_USERS_INITIAL) {
       const newFollowedUsers = [...initialUsers];
       exhangeFollower(newFollowedUsers, willFollow, index);
-
       setInitialUsers(newFollowedUsers);
+      if (!netInfo.isConnected)
+        setTimeout(() => {
+          exhangeFollower(newFollowedUsers, !willFollow, index);
+          setInitialUsers(newFollowedUsers);
+        }, 2000);
     }
 
-    const data = {
-      user_id_follower: myId,
-      user_id_followed: item.user_id,
-      username_follower: profile.myProfile.username,
-      username_followed: item.username,
-      follow_source: 'discoveryScreen'
-    };
-
-    if (willFollow) {
-      await setFollow(data, client);
-    } else {
-      await setUnFollow(data, client);
+    if (netInfo.isConnected) {
+      const data = {
+        user_id_follower: myId,
+        user_id_followed: item.user_id,
+        username_follower: profile.myProfile.username,
+        username_followed: item.username,
+        follow_source: 'discoveryScreen'
+      };
+
+      if (willFollow) {
+        await setFollow(data, client);
+      } else {
+        await setUnFollow(data, client);
+      }
+      if (searchText.length > 0) fetchData();
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a delay of 2000ms (2 seconds) when the network is not connected. This delay seems to be used to revert the follower status if the network is still disconnected after the delay. However, this approach could lead to inconsistent states if the network connection is restored during the delay period. It would be better to handle network disconnection more gracefully, perhaps by showing an error message to the user and preventing them from changing the follower status until the network is reconnected.

```diff
-      if (!netInfo.isConnected)
-        setTimeout(() => {
-          exhangeFollower(newFollowedUsers, !willFollow, index);
-          DiscoveryAction.setDiscoveryInitialUsers(newFollowedUsers, discoveryDispatch);
-        }, 2000);

+      if (!netInfo.isConnected) {
+        // Show an error message to the user
+        Alert.alert("Network Error", "Please check your internet connection and try again.");
+        return;
+      }
```

This change should be applied to all similar instances in the code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->